# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../pdisk/code_update.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# Copyright (C) 1998 - 2003 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./errors.c	2000-05-16 09:10:57.000000000 -0500
+++ ./errors.c	2002-12-19 20:51:14.000000000 -0500
@@ -29,21 +29,24 @@
 #include <stdio.h>
 
 // for exit()
-#ifndef __linux__
 #include <stdlib.h>
-#endif
-// for strrchr
+
+// for strrchr and strerror
 #include <string.h>
 
 // for va_start(), etc.
 #include <stdarg.h>
 
+// for errno
+#include <errno.h>
+
 #include "errors.h"
 
 
 //
 // Defines
 //
+#define sys_nerr errno
 
 
 //
@@ -134,7 +137,7 @@
 
 #if defined(__linux__) || defined(NeXT) || defined(__unix__)
     if (value > 0 && value < sys_nerr) {
-	fprintf(stderr, "  (%s)\n", sys_errlist[value]);
+	fprintf(stderr, "  (%s)\n", strerror(value));
     } else {
 	fprintf(stderr, "\n");
     }
@@ -163,7 +166,7 @@
 
 #if defined(__linux__) || defined(NeXT) || defined(__unix__)
     if (value > 0 && value < sys_nerr) {
-	fprintf(stderr, "  (%s)\n", sys_errlist[value]);
+	fprintf(stderr, "  (%s)\n", strerror(value));
     } else {
 	fprintf(stderr, "\n");
     }
--- ./file_media.c	2000-05-16 09:10:55.000000000 -0500
+++ ./file_media.c	2002-12-19 20:48:18.000000000 -0500
@@ -58,6 +58,7 @@
  * Defines
  */
 #ifdef __linux__
+#define llseek lseek64
 #define LOFF_MAX 9223372036854775807LL
 extern __loff_t llseek __P ((int __fd, __loff_t __offset, int __whence));
 #else
